ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -583,7 +583,15 @@ def _close(self):
         self._loaded = False
         self.communicator.close()
         if self.proc1 is not None:
-            self.proc1.kill()
+            # Wait a few seconds for the process to shutdown, but kill it if it takes too long
+            try:
```

Comment:
This seems necessary (but not sufficient) to the environment cleanly shutting down. Might move to another PR.